app/src/main/java/fr/free/nrw/commons/auth/login/LoginClient.kt

```diff
@@ -106,6 +107,35 @@ class LoginClient(private val loginInterface: LoginInterface) {
         })
     }
 
+    fun doLogin(
+        username: String,
+        password: String,
+        twoFactorCode: String? = null,
+        userLanguage: String = Locale.getDefault().language,
+        loginCallback: LoginCallback
+    ) {
+        getLoginToken().enqueue(object :Callback<MwQueryResponse?>{
+            override fun onResponse(
+                call: Call<MwQueryResponse?>,
+                response: Response<MwQueryResponse?>
+            ) {
+                if (response.isSuccessful){
+                    val loginToken = response.body()?.query()?.loginToken()
+                    loginToken?.let {
+                        login(username, password, null, twoFactorCode, it, userLanguage, loginCallback)
+                    } ?: run {
+                        loginCallback.error(IOException("Failed to retrieve login token"))
+                    }
+                } else {
+                        loginCallback.error(IOException("Failed to retrieve login token"))
```

Comment:
Not a big deal, but the indenting seems off 😄

Reply:
fixed the indentation. I hope adding klint will fix such issues before commit.